packages/application-config/src/formatters.ts

```diff
@@ -55,6 +56,22 @@ const formatEntryPointUriPathToResourceAccessKey = (
     })
     .join('');
 
+/**
+ * The function formats the `permissionName` to a resource access key.
+ * It makes the first character of the string and the next character after a special character an uppercase.
+ *
+ * @example
+ * - avengers --> Avengers
+ * - the-avengers --> TheAvengers
+ * - avengers-01 --> Avengers01 [ℹ️ mind the difference here compared to `formatEntryPointUriPathToResourceAccessKey`] */
+const formatPermissionNameToResourceAccessKey = (permissionName: string) =>
+  permissionName
+    // Each word is split by a hyphen.
+    .split('-')
+    .map((word) => word.toLowerCase())
```

Comment:
I think we can omit this, this should be covered by the validations.

Reply:
Ah, got you :smile: